Lib/multiprocessing/shared_memory.py

```diff
@@ -0,0 +1,573 @@
+        pass
+
+
+class PosixSharedMemory(_PosixSharedMemory):
+
+    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
+        if name and (flags is None):
+            _PosixSharedMemory.__init__(self, name)
+        else:
+            if name is None:
+                name = f'psm_{os.getpid()}_{random.randrange(100000)}'
+            _PosixSharedMemory.__init__(self, name, flags=O_CREX, size=size)
+
+        self._mmap = mmap.mmap(self.fd, self.size)
+        self.buf = memoryview(self._mmap)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.name!r}, size={self.size})'
+
+    def close(self):
+        self.buf.release()
+        self._mmap.close()
+        self.close_fd()
+
+
+class SharedMemory:
+
+    def __new__(cls, *args, **kwargs):
+        if os.name == 'nt':
+            cls = WindowsNamedSharedMemory
+        else:
+            cls = PosixSharedMemory
+        return cls(*args, **kwargs)
+
+
+def shareable_wrap(
+    existing_obj=None,
+    shmem_name=None,
+    cls=None,
+    shape=(0,),
+    strides=None,
+    dtype=None,
+    format=None,
+    **kwargs
+):
+    augmented_kwargs = dict(kwargs)
+    extras = dict(shape=shape, strides=strides, dtype=dtype, format=format)
+    for key, value in extras.items():
+        if value is not None:
+            augmented_kwargs[key] = value
+
+    if existing_obj is not None:
+        existing_type = getattr(
+            existing_obj,
+            "_proxied_type",
+            type(existing_obj)
+        )
+
+        #agg = existing_obj.itemsize
+        #size = [ agg := i * agg for i in existing_obj.shape ][-1]
+        # TODO: replace use of reduce below with above 2 lines once available
+        size = reduce(
+            lambda x, y: x * y,
+            existing_obj.shape,
+            existing_obj.itemsize
+        )
+
+    else:
+        assert shmem_name is not None
+        existing_type = cls
+        size = 1
+
+    shm = SharedMemory(shmem_name, size=size)
+
+    class CustomShareableProxy(existing_type):
+
+        def __init__(self, *args, buffer=None, **kwargs):
+            # If copy method called, prevent recursion from replacing _shm.
+            if not hasattr(self, "_shm"):
+                self._shm = shm
+                self._proxied_type = existing_type
+            else:
+                # _proxied_type only used in pickling.
+                assert hasattr(self, "_proxied_type")
+            try:
+                existing_type.__init__(self, *args, **kwargs)
+            except:
+                pass
+
+        def __repr__(self):
+            if not hasattr(self, "_shm"):
+                return existing_type.__repr__(self)
+            formatted_pairs = (
+                "%s=%r" % kv for kv in self._build_state(self).items()
+            )
+            return f"{self.__class__.__name__}({', '.join(formatted_pairs)})"
+
+        #def __getstate__(self):
```

Comment:
This is commented-out code and should be removed.

Reply:
Removed in GH-11816.